### PR TITLE
implement #12231 new flag --xfail-tb

### PR DESCRIPTION
**Pull request by okken**
This implements issue #12231, making xfail tracebacks optional.   
Introduces a new flag --xfail-tb, that acts just like --tb, but just for xfail tracebacks.  

With this feature in place, turning off xfail tracebacks is as simple as `--xfail-tb=no`, and even works when `-rx` or `-ra` are present.
   
This is a first pass at the PR.  
- One test has been modified. 

Still todo:                                     
- Modify documentation for new flag
- Create changelog entry 
- I think that's it ??

### CODE REVIEW

File: src/_pytest/terminal.py [hunk 1]
```diff
@@ -216,6 +216,15 @@ def pytest_addoption(parser: Parser) -> None:
         choices=["auto", "long", "short", "no", "line", "native"],
         help="Traceback print mode (auto/long/short/line/native/no)",
     )
+    group._addoption(
+        "--xfail-tb",
+        metavar="style",
+        action="store",
+        dest="xfail_tbstyle",
+        default="auto",
```
Comment (nicoddemus) [inline src/_pytest/terminal.py:224]: I think we should use `no` as default in this case, to get the pre-8.1.0 behavior, as people has been side tracked by this a lot (including myself at work, where we have some xfail in places).

```suggestion
        default="no",
```

We will need to mention this decision very clearly in the changelog.

File: src/_pytest/terminal.py [hunk 2]
```diff
@@ -1074,18 +1083,25 @@ def summary_failures(self) -> None:
...
+        style = self.config.option.xfail_tbstyle
+        self.summary_failures_combined("xfailed", "XFAILURES", "x", style=style)
 
     def summary_failures_combined(
-        self, which_reports: str, sep_title: str, needed_opt: Optional[str] = None
+        self,
+        which_reports: str,
+        sep_title: str,
+        needed_opt: Optional[str] = None,
+        style=None,
```
Comment (nicoddemus) [inline src/_pytest/terminal.py:1094]:
```suggestion
        style: Optional[str]=None,
```

**Review by nicoddemus — CHANGES_REQUESTED** [review]
Thanks @okken for tackling this so quickly!

As you mention indeed we still need tests and docs for this, but also I would like to see some other specific tests:

1. a normal failure and an xfail failure, with `tb=line` and `xfail-tb=auto`, and check the output.
2. I suppose just changing `--xfail-tb` default to `no` will require update some test(s), but if not, we need to check that the default does not produce any xfail tracebacks.

**Comment by okken** [comment]
I've written a rather verbose changelog entry. Feel free to shorten, tighten, completely rewrite, make suggestions, etc.

**Comment by okken** [comment]
@nicoddemus I think I've done all necessary changes

**Comment by okken** [comment]
I didn't quite do the tests as requested. by "a normal failure and an xfail failure, with tb=line and xfail-tb=auto, and check the output."
1. I didn't see the "a normal failure and" part until just now. I can add that.
2. None of the --tb tests use auto, but use short. So I did that.

Right now I've got a -rx test, an -rx --xfail-tb=short, and an -rx --xfail-tb=line test.

I think that's sufficient.
But if you'd like me to add the "and normal failure" part to the test, I can do that also.

**Comment by okken** [comment]
Converting to draft, because I'm still working on it, and thinking about behavior. see #12231 for discussion and questions

**Comment by okken** [comment]
Accidentally deleted my fork, along with this branch. 
I've recreated the PR as #12280

**Pull request closed without merging**